src/frontend/src/Components/map/Map.js

```diff
@@ -753,6 +782,7 @@ export default function DriveBCMap(props) {
           onClick={() => {
             if (referenceData) {
               setZoomPan(mapView, 12, fromLonLat(referenceData.location.coordinates));
+              highlighClickedCamera();
```

Comment:
At this point we should already have the 'referenceFeature' available as the feature that was passed in; there is no need to do lookup for it again. Simply calling setStyle with it here will suffice.

Reply:
I tried calling setStyle with the referenceFeature, and while it works to highlight the referenceFeature when available, we still encounter a few issues that need to be addressed:

Unhighlighting Other Cameras: We also need to ensure that other cameras are unhighlighted when the referenceFeature is highlighted. This behavior is necessary to prevent multiple cameras from being highlighted simultaneously.

No Camera Selected: When the user clicks on an area with no camera selected, the referenceFeature becomes null, which causes the issue because we can't apply a style to a non-existent feature.

Camera Orientation Changes: The issue also arises when the user switches camera orientations within a group. The referenceFeature approach does not handle this case correctly, as it relies on the referenceFeature that might not be properly updated.

Given these scenarios, I believe it's safer to keep the lookup function to handle all these edge cases and ensure consistent behavior across the different interactions.